manta-crypto/src/arkworks/pairing.rs

```diff
@@ -106,6 +111,177 @@ pub trait PairingEngineExt: PairingEngine {
 
 impl<E> PairingEngineExt for E where E: PairingEngine {}
 
+/// Generates the `G2Prepared` and `G2PreparedRef` structures for serialization compatibility
+/// with arkworks canonical serialization.
+macro_rules! pairing_impl {
+    ($params:ident) => {
+        /// Line Evaluation Coefficients
+        pub type EllCoeff<F> = (F, F, F);
+
+        /// G2 Prepared Point
+        #[derive(derivative::Derivative, CanonicalSerialize, CanonicalDeserialize)]
+        #[derivative(Clone, Default, Debug, Eq, PartialEq)]
+        pub struct G2Prepared<P>
```

Comment:
If we have three (projective) coordinates, then the infinity flag makes no sense. If we had two (affine) coordinates, then this should probably be an enum and not a struct.

Reply:
This `struct` is copied directly from `arkworks`, we didn't decide how it was written.